test/cmdLineTests/jvmtitests/src/com/ibm/jvmti/tests/nestMatesRedefinition/ClassGenerator.java

```diff
@@ -0,0 +1,197 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package com.ibm.jvmti.tests.nestMatesRedefinition;
+
+import org.objectweb.asm.*;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+
+
+public class ClassGenerator implements Opcodes {
+
+	public static byte[] nestHostNoAttribute () throws Exception {
+		ClassWriter classWriter = new ClassWriter(0);
```

Comment:
**Relevant to later ClassGenerator format comments

When creating the class generator, I used the ASMifier asm utility to read in a compiled .class and produce the asm code to generate it. I then manually added nest host and nestmates attributes and neatened the code (indenting, removing unused variables). However, I had not changed constructor arguments, the location of variable definitions, or scopes. Will correct these now.

Reply:
When compiling locally, the compiler didn't recognize `COMPUTE_FRAMES` and `COMPUTE_MAXS` flags - left it as-is.

Reply:
Did you reference them with a static import or by `ClassWriter.COMPUTE_FRAMES`?

Reply:
Updating with ClassWriter.COMPUTE_MAXS now.

Choosing COMPUTE_MAXS over COMPUTE_FRAMES because the classwriter constructor only uses one of them and it checks for the COMPUTE_FRAMES flag first and the class generator code here uses `methodVisitor.visitMaxs` (flag passed from classwriter constructor to methodvisitor constructor)

Reply:
The two values can be `|` together and both passed in.  I'm pretty sure you'll find that pattern throughout the test code.